google-cloud-core/src/main/java/com/google/cloud/HttpTransportOptions.java

```diff
@@ -186,7 +186,10 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object obj) {
-    if (!(obj instanceof HttpTransportOptions)) {
+    if (obj == null) {
```

Comment:
I'll leave this for another day. I doubt people are doing much comparison of these.